clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2293,6 +2321,33 @@ defaults to ``1``.
 In OpenCL C, this attribute is available in GNU spelling
 (``__attribute__((reqd_work_group_size(X, Y, Z)))``), see section
 6.7.2 Optional Attribute Qualifiers of OpenCL 1.2 specification for details.
+
+.. code-block:: c++
+
+  __kernel __attribute__((reqd_work_group_size(8, 16, 32))) void test() {}
+
+  [[cl::reqd_work_group_size(4, 4, 4)]] void foo() {}
+
+  class Foo {
+  public:
+    [[cl::reqd_work_group_size(2, 2, 2)]] void operator()() const {}
+  };
+
+  template <int N, int N1, int N2>
+  class Functor {
+  public:
+    [[cl::reqd_work_group_size(N, N1, N2)]] void operator()() const {}
+  };
+
+  template <int N, int N1, int N2>
+  [[cl::reqd_work_group_size(N, N1, N2)]] void func() {}
+
+  [[intel::reqd_work_group_size(N, N1, N2)]] void bar() {}
+
+  ``[[intel::reqd_work_group_size(N, N1, N2)]]`` takes two optional
```

Comment:
Isn't this valid in other spellings as well? Not just `intel::`?

Reply:
Not understand your question here @elizabethandrews .  Did you mean to say about optional parameters?
 // If no attribute argument is specified, set to default value '1'
  // for second and third attribute argument in ReqdWorkGroupSizeAttr
  // for only with intel::reqd_work_group_size spelling.

Reply:
Yes. Sorry for not being more clear. I was wondering if the line about optional arguments applied to other spellings as well.

Reply:
Thanks for the clarification. No, it only applied to spelling intel::

// If no attribute argument is specified, set to default value '1'
// for second and third attribute argument in ReqdWorkGroupSizeAttr
// for only with intel::reqd_work_group_size spelling.

Reply:
So for other spellings, N1 and N2 are not optional? Do you know why this discrepancy exists? Should we also mention this in documentation?

Reply:
>> So for other spellings, N1 and N2 are not optional?

Yes, correct.

>>Do you know why this discrepancy exists? 

No, i do not know why. From git blame or source codes i learnt that the intel:: was originally implemented with this optional parameters. 

We have already mentioned this in Attr.td file:
 // Default arguments in ReqWorkGroupSize can be used only with
// intel::reqd_work_group_size spelling.
def ReqdWorkGroupSize : InheritableAttr {

Also in AttrDocs.file here too:
As Intel extension, ``[[intel::reqd_work_group_size(X, Y, Z)]]`` spelling is allowed
which features optional arguments `Y` and `Z`, those simplifies its usage if
only 1- or 2-dimensional ND-range is assumed by a programmer. These arguments
defaults to ``1``.

Reply:
I have moved the examples in a separate block for each attribute spelling to make it more clear.